wp_insert_rows.php

```diff
@@ -76,10 +80,28 @@ function wp_insert_rows($row_arrays = array(), $wp_table_name) {
 	
 	$query .= implode(', ', $place_holders);
 	
-	if($wpdb->query($wpdb->prepare($query, $values))){
-	    return true;
-	} else {
-	    return false;
-	}
+  if ($update)	
+  {
+    $update=" ON DUPLICATE KEY UPDATE $prikeyname=VALUES( $prikeyname ),";
+    $cnt=0;
+    foreach($row_arrays[0] as $key => $value)
+    {
+      if($cnt==0) {
+        $update .= "$key=VALUES($key)";
+        $cnt=1;
+      }
+      else {
+        $update .= ", $key=VALUES($key)";
+      }
+    }
+    $query .= $update;
+  }
+  $sql=$wpdb->prepare($query, $values);
+
+  if($wpdb->query($sql)){
```

Comment:
If there isn't any specific reason please follow the old code style -> if($wpdb->query($wpdb->prepare($query, $values))){